### PR TITLE
SHARE-539 Improve feedback for download project button

### PR DESCRIPTION
**Pull request by EifonUser**
---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [x] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [x] Choose the proper base branch (*develop*)
- [ ] Confirm that the changes follow the project’s coding guidelines
- [ ] Verify that the changes generate no warnings and errors 
- [ ] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Verify that all tests are passing (CI), if not please state the test cases in the [section](#Tests) below
- [x] Perform a self-review of the changes
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [ ] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer
- [ ] Check that your pull request has been successfully deployed to https://web-test-1.catrob.at/

### Additional Description
SweetAlert is commented out, because I felt like it makes the user experience worse. So it can be added if still needed.

### Tests - additional information
`TODO: add additional information about testruns here`

### CODE REVIEW

File: assets/js/custom/Program.js [hunk 1]
```diff
@@ -45,6 +45,13 @@ export const Program = function (projectId, projectName, userRole, myProgram, st
     )
   })
 
+  $($('.js-btn-project-download-disabled').on('click', (e) => {
+    downloadDisabled(
+      $(e.currentTarget).data('redirect-url')
+      // $(e.currentTarget).data('alert-text')
```
Comment (dmetzner) [inline assets/js/custom/Program.js:51]: please delete the commented-out code everywhere

File: src/Application/Controller/Project/ProgramController.php [hunk 2]
```diff
@@ -143,7 +145,7 @@ public function projectLikeAction(Request $request, string $id): Response
     $user = $this->getUser();
     if (!$user) {
       if ($request->isXmlHttpRequest()) {
-        return new JsonResponse(['statusCode' => 601], Response::HTTP_UNAUTHORIZED);
+        return new JsonResponse(['statusCode' => 401], Response::HTTP_UNAUTHORIZED);
```
Comment (dmetzner) [inline src/Application/Controller/Project/ProgramController.php:148]: Would be nice if you could remove the statusCode fully.

Comment (dmetzner) [reply src/Application/Controller/Project/ProgramController.php:148]: (however not part of your task - so no need in this ticket)

File: templates/Program/program_download_disabled_button.html.twig [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    {% include 'components/loading_spinner.html.twig' with {'spinner_id': 'download-disabled-progressbar' ~ suffix|default(), 'size': 'small'} only %}
+  </i>
+
+  {# Icon #}
+  <i id="projectDownloadDisabledButton__icon{{ suffix|default('') }}" class="material-icons align-bottom">lock</i>
+
+  {# Text #}
+  <span>{{ "download"|trans({}, "catroweb") }}</span>
+
+</button>
```
Comment (dmetzner) [inline templates/Program/program_download_disabled_button.html.twig:18]: is really a new template needed to disable a button? maybe think about adding a different variable to make the existing template more flexible and reduce the duplicated code

**Review by dmetzner — COMMENTED** [review]
Looks good to me. Just a few minor inputs from my side :)

File: templates/Program/program_download_disabled_button.html.twig (hunk 3 above)
Comment (EifonUser) [reply templates/Program/program_download_disabled_button.html.twig:18]: I was thinking about the same but if you add a new variable instead, you could go into dev tools and change it form eg 0 to 1 and by doing that you could enable the download button, even if you are not logged in.

Comment (dmetzner) [reply templates/Program/program_download_disabled_button.html.twig:18]: The request is locked anyway to users not logged in, so the user would not achieve anything by doing such things. However, it's also okay this way; I mean, those are not even 20 lines of code. all good -just wanted to ask to get your why :)

File: assets/js/custom/Program.js (hunk 1 above)
Comment (dmetzner) [reply assets/js/custom/Program.js:51]: can be done in a new PR

**Review by dmetzner — APPROVED** [review]

**Pull request merged**